RunCat365/StartupManager.cs

```
@@ -0,0 +1,145 @@
+            {
+                return SetStartupAsync_Packed(isChecked).Result;
+            }
+            else
+            {
+                return SetStartup_NotPacked(isChecked);
+            }
+        }
+
+        private static async Task<bool> GetStartupAsync_Packed()
+        {
+            if (startupTask is null) startupTask = await StartupTask.GetAsync("RunCatStartup");
+            if (startupTask is null) return false;
+            if (startupTask.State == StartupTaskState.Enabled) return true;
+            return false;
+        }
+
+        private static bool GetStartup_NotPacked()
+        {
+            var keyName = @"Software\Microsoft\Windows\CurrentVersion\Run";
+            using var rKey = Registry.CurrentUser.OpenSubKey(keyName);
+            if (rKey is null) return false;
+            var value = (rKey.GetValue(Application.ProductName) is not null);
+            rKey.Close();
+            return value;
+        }
+
+        private static async Task<bool> SetStartupAsync_Packed(bool isChecked)
+        {
+            var active = !isChecked;
+            var changeCheck = false;
+            if (startupTask is null) startupTask = await StartupTask.GetAsync("RunCatStartup");
+            if (active)
+            {
+                switch (startupTask.State)
+                {
+                    case StartupTaskState.Enabled:
+                        changeCheck = true;
+                        break;
+                    case StartupTaskState.Disabled:
+                        StartupTaskState newState = await startupTask.RequestEnableAsync();
+                        if (newState == StartupTaskState.Enabled)
+                        {
+                            changeCheck = true;
+                        }
+                        else
+                        {
+                            MessageBox.Show("Launch at Startup could not be activated.", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
```

Instead of calling `MessageBox.Show()` here, it would be better to throw an error and handle it in the `ContextMenuManager` that uses `SetStartup`. 
Since `ContextMenuManager` deals with UI-related operations, it makes more sense there.